ledger-core/virtual/integration/constructor_test.go

```diff
@@ -123,7 +125,94 @@ func TestVirtual_Constructor_WithExecutor(t *testing.T) {
 
 	server.SendPayload(ctx, &pl)
 
-	assert.True(t, server.PublisherMock.WaitCount(1, 10*time.Second))
+	{
+		select {
+		case <-executeDone:
+		case <-time.After(10 * time.Second):
+			require.FailNow(t, "timeout")
+		}
+
+		select {
+		case <-server.Journal.WaitAllAsyncCallsDone():
+		case <-time.After(10 * time.Second):
+			require.FailNow(t, "timeout")
+		}
+	}
+	assert.Equal(t, 1, typedChecker.VCallResult.Count())
+
+	mc.Finish()
+}
+
+func TestVirtual_Constructor_CurrentPulseWithoutObject(t *testing.T) {
+	t.Log("C4995")
+
+	var (
+		mc        = minimock.NewController(t)
+		isolation = contract.ConstructorIsolation()
+		class     = gen.UniqueReference()
+	)
+
+	server, ctx := utils.NewUninitializedServer(nil, t)
+	defer server.Stop()
+
+	executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
+
+	runnerMock := logicless.NewServiceMock(ctx, mc, nil)
+	server.ReplaceRunner(runnerMock)
+	server.Init(ctx)
+
+	var (
+		outgoing     = server.RandomLocalWithPulse()
+		objectRef    = reference.NewSelf(outgoing)
+		runnerResult = []byte("123")
+	)
+
+	pl := payload.VCallRequest{
+		CallType:       payload.CTConstructor,
+		CallFlags:      payload.BuildCallFlags(isolation.Interference, isolation.State),
+		CallAsOf:       server.GetPulse().PulseNumber,
+		Callee:         class,
+		CallSiteMethod: "test",
+		CallOutgoing:   outgoing,
+	}
+	msg := server.WrapPayload(&pl).Finalize()
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+	typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+		require.Equal(t, res.ReturnArguments, runnerResult)
+		require.Equal(t, res.Callee, objectRef)
+		require.Equal(t, res.CallOutgoing, outgoing)
+
+		return false // no resend msg
```

Comment:
check CTMethod and CallFlags , too, please

Reply:
done

Reply:
done